lib/logstash/inputs/elasticsearch.rb

```diff
@@ -126,6 +127,22 @@ class LogStash::Inputs::Elasticsearch < LogStash::Inputs::Base
   # by this pipeline input.
   config :slices, :validate => :number
 
+  # Enable tracking the value of a given field to be used as a cursor
+  # TODO: main concerns
+  #       * schedule overlap needs to be disabled (hardcoded as enabled)
+  #       * using anything other than _event.timestamp easily leads to data loss
+  #       * the first "synchronization run can take a long time"
+  #       * checkpointing is only safe to do after each run (not per document)
+  config :tracking_field, :validate => :string
+
+  # Define the initial seed value of the tracking_field
+  config :tracking_field_seed, :validate => :string
+
+  # The location of where the tracking field value will be stored
+  # The value is persisted after each scheduled run (and not per result)
+  # If it's not set it defaults to '${path.data}/plugins/inputs/elasticsearch/last_run_value'
+  config :last_run_metadata_path, :validate => :string
```

Comment:
Might be good to call out in documentation that this needs to be configured explicitly with a unique path if you use more than one `elasticsearch` input within your pipeline, otherwise they'll share the same tracking file.

Reply:
Indeed, we can harden this by opening the file in exclusive mode + document the constraint.

Reply:
I've changed the plugin to store the file in a folder namespaced by the pipeline id:

```asciidoc
[id="plugins-{type}s-{plugin}-last_run_metadata_path"]
===== `last_run_metadata_path`

  * Value type is <<string,string>>
  * There is no default value for this setting.

The path to store the last observed value of the tracking field, when used.
By default this file is stored as `<path.data>/plugins/inputs/elasticsearch/<pipeline_id>/last_run_value`.

This setting should point to file, not a directory, and Logstash must have read+write access to this file.
```